lib/oauth.rb

```diff
@@ -238,16 +239,15 @@ def encode_uri_component(uri_component)
     #
     def get_nonce(len = 32)
       # Returns a random string of length=len
-      o = [('a'..'z'), ('A'..'Z'), (0..9)].map(&:to_a).flatten
-      (0...len).map { o[rand(o.length)] }.join
+      SecureRandom.alphanumeric(len)
     end
 
     # Returns UNIX Timestamp as required per
     # https://tools.ietf.org/html/rfc5849#section-3.3
     # @return {String} UNIX timestamp (UTC)
     #
     def time_stamp
-      Time.now.getutc.to_i
+      Time.now.to_i
```

Comment:
Is there a specific reason for removing `getutc`?

Reply:
Yes, its is not necessary. A unix timestamp is always based on UTC - calling `getutc` is unnecessary extra work.

```ruby
puts "#{Time.now.zone}: #{Time.now.to_i}\n#{Time.now.getutc.zone}: #{Time.now.getutc.to_i}"
#=> AEST: 1694001822
#=> UTC: 1694001822
```